Source/DafnyLanguageServer.Test/Util/AssertWithDiff.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Text;
+using DiffPlex.DiffBuilder;
+using DiffPlex.DiffBuilder.Model;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Dafny.LanguageServer.IntegrationTest.Util;
+
+public class AssertWithDiff {
+  public static void Equal(string expected, string actual) {
+    var diff = InlineDiffBuilder.Instance.BuildDiffModel(expected, actual);
+    if (!diff.HasDifferences) {
+      return;
+    }
+
+    var message = new StringBuilder();
+    message.AppendLine("AssertEqualWithDiff() Failure");
+    message.AppendLine("Diff (changing expected into actual):");
+    foreach (var line in diff.Lines) {
+      var prefix = line.Type switch {
+        ChangeType.Inserted => '+',
+        ChangeType.Deleted => '-',
+        _ => ' '
+      };
+      message.Append(prefix);
+      message.AppendLine(line.Text);
+    }
+
+    Assert.AreEqual(expected, actual, message.ToString());
```

Comment:
This file is the same as in XUnitExtensions's AssertWithDiff excepted for this line which uses the framework that the DafnyLanguageServer relies on. I can't factor them both in a single file because we don't have yet a project on which all test projects depend, and I don't think this PR should take care of that.